### ISSUE TITLE
spack install gcc fails

### ISSUE THREAD

**Issue by behlendorf** [LLNL/spack#1893]
Using version v0.8.17-4042-g50fcade (50fcade) I encountered the following failure when trying to install gcc in a stock Amazon Linux instance.

```
$ git clone https://github.com/LLNL/spack.git
Cloning into 'spack'...
remote: Counting objects: 43408, done.
remote: Compressing objects: 100% (90/90), done.
remote: Total 43408 (delta 26), reused 2 (delta 2), pack-reused 43303
Receiving objects: 100% (43408/43408), 10.38 MiB | 12.81 MiB/s, done.
Resolving deltas: 100% (21313/21313), done.
Checking connectivity... done.

$ cd spack/
$ ./bin/spack install gcc
==> Installing gcc
==> Installing gmp
==> Installing m4
==> Installing libsigsegv
==> Trying to fetch from file:///home/ec2-user/spack/var/spack/cache/libsigsegv/libsigsegv-2.10.tar.gz
curl: (37) Couldn't open file /home/ec2-user/spack/var/spack/cache/libsigsegv/libsigsegv-2.10.tar.gz
==> Fetching from file:///home/ec2-user/spack/var/spack/cache/libsigsegv/libsigsegv-2.10.tar.gz failed.
==> Trying to fetch from ftp://ftp.gnu.org/gnu/libsigsegv/libsigsegv-2.10.tar.gz
######################################################################## 100.0%
==> Staging archive: /home/ec2-user/spack/var/spack/stage/libsigsegv-2.10-4ztqjpi2ehos3tf54udul4czzkykpiej/libsigsegv-2.10.tar.gz
==> Created stage in /home/ec2-user/spack/var/spack/stage/libsigsegv-2.10-4ztqjpi2ehos3tf54udul4czzkykpiej
==> No patches needed for libsigsegv
==> Building libsigsegv
==> Successfully installed libsigsegv
  Fetch: 4.78s.  Build: 6.31s.  Total: 11.10s.
[+] /home/ec2-user/spack/opt/spack/linux--x86_64/gcc-4.8.3/libsigsegv-2.10-4ztqjpi2ehos3tf54udul4czzkykpiej
Traceback (most recent call last):
  File "./bin/spack", line 192, in <module>
    main()
  File "./bin/spack", line 169, in main
    return_val = command(parser, args)
  File "/home/ec2-user/spack/lib/spack/spack/cmd/install.py", line 92, in install
    explicit=True)
  File "/home/ec2-user/spack/lib/spack/spack/package.py", line 934, in do_install
    dirty=dirty)
  File "/home/ec2-user/spack/lib/spack/spack/package.py", line 1085, in do_install_dependencies
    dep.package.do_install(**kwargs)
  File "/home/ec2-user/spack/lib/spack/spack/package.py", line 934, in do_install
    dirty=dirty)
  File "/home/ec2-user/spack/lib/spack/spack/package.py", line 1085, in do_install_dependencies
    dep.package.do_install(**kwargs)
  File "/home/ec2-user/spack/lib/spack/spack/package.py", line 934, in do_install
    dirty=dirty)
  File "/home/ec2-user/spack/lib/spack/spack/package.py", line 1085, in do_install_dependencies
    dep.package.do_install(**kwargs)
  File "/home/ec2-user/spack/lib/spack/spack/package.py", line 1057, in do_install
    self.spec, spack.install_layout, explicit=explicit)
  File "/home/ec2-user/spack/lib/spack/spack/database.py", line 78, in converter
    return function(self, spec_like, *args, **kwargs)
  File "/home/ec2-user/spack/lib/spack/spack/database.py", line 512, in add
    self._add(spec, directory_layout, explicit=explicit)
  File "/home/ec2-user/spack/lib/spack/spack/database.py", line 475, in _add
    except DirectoryLayoutError as e:
NameError: global name 'DirectoryLayoutError' is not defined
```

**Comment by tgamblin** [comment]
@behlendorf: There was a missing import in an edge case.  I am hoping this was just a syntax error but I worry that something else was actually raised to trigger that error.  Can you try it again and see if it works now?  Unfortunately I don't have package testing set up on AWS yet, or I would try it out.

**Comment by behlendorf** [comment]
@tgamblin thanks for the speedy fix.  That resolved this issue, I've opened #1899 with the next failure I observed.
